nengo_spinnaker/operators/filter.py

```diff
@@ -19,252 +19,288 @@
+        )
+
+        # Build all the vertices
+        self.cores = [
+            FilterCore(self.column_slice, out_slice,
+                       transform_region, output_keys, output_slices,
+                       machine_timestep,
+                       filter_region, filter_routing_region) for
+            out_slice in divide_slice(slice(0, size_out), n_cores)
+        ]
+
+        return self.cores
+
+    def load_to_machine(self, netlist, controller):
+        """Allocate shared memory on each chip that we're using and load data
+        to SDRAM for each core that we have allocated.
+        """
+        # Get each core to load itself
+        for core in self.cores:
+            core.load_to_machine(netlist)
+
+
+class FilterCore(Vertex):
+    """Portion of the rows of the transform assigned to a parallel filter
+    group, represents the load assigned to a single processing core.
     """
-    def __init__(self, in_slice, out_slice, resources=dict(),
-                 transmission_parameter_slices=list()):
-        super(ParallelFilterSlice, self).__init__(get_application("filter"),
-                                                  resources)
+    def __init__(self, column_slice, output_slice,
+                 transform_region, output_keys, output_slices,
+                 machine_timestep, filter_region, filter_routing_region):
+        """Allocate a portion of the overall matrix to a single processing
+        core.
 
-        # Store the slices
-        self.in_slice = in_slice
-        self.out_slice = out_slice
+        Parameters
+        ----------
+        column_slice : :py:class:`slice`
+            Columns of the transform matrix managed by the group of vertices of
+            which we are a member.
+        output_slice : :py:class:`slice`
+            Slice of the rows of the transform matrix that will be applied by
+            this processing core.
+        transform_region : MatrixRegion
+        output_keys : [BitField, ...]
+            Keys transmitted by filter.
+        output_slices : [(TransmissionParameters, set), ...]
+            Pairs of transmission parameters and sets containing the row
+            indices of the transform matrix corresponding to the transmission
+            parameters.
+        """
+        # Store information about the slices of the for which matrix we're
+        # responsible.
+        self.output_slice = output_slice
+        self.column_slice = column_slice
 
         # Store which signal parameter slices we contain
         self.transmission_params = list()
-        out_set = set(range(out_slice.start or 0,
-                            out_slice.stop or 0,
-                            out_slice.step or 1))
-        for transmission_params, outs in transmission_parameter_slices:
+        out_set = set(range(output_slice.start or 0,
+                            output_slice.stop or 0,
+                            output_slice.step or 1))
```

Comment:
I don't really understand what the default values for stop is for? Are slices even valid without stop and wouldn't this default leave the core not transmitting anything?